scripts/provisioning/s3_haproxy_config.py

```diff
@@ -59,6 +59,42 @@ def __init__(self, confstore: str):
     self.machine_id = self.provisioner_confstore.get_machine_id()
     self.logger.info(f'Machine id : {self.machine_id}')
 
+  def parse_endpoint(self, endpoint_str):
+    """Parse endpoint string and return dictionary with components:
+         * scheme,
+         * fqdn,
+         * and optionally port, if present in the string.
+
+       Examples:
+
+       'https://s3.seagate.com:443' -> {'scheme': 'https', 'fqdn': 's3.seagate.com', 'port': '443'}
+       'https://s3.seagate.com'     -> {'scheme': 'https', 'fqdn': 's3.seagate.com'}
+       'http://s3.seagate.com:80'   -> {'scheme': 'http', 'fqdn': 's3.seagate.com', 'port': '80'}
+       'http://127.0.0.1:80'        -> {'scheme': 'http', 'fqdn': '127.0.0.1', 'port': '80'}
+    """
+    try:
+      result1 = urllib.parse.urlparse(endpoint_str)
```

Comment:
add `import urllib.parse` at the top of the file.

Reply:
Done.